### ISSUE TITLE
QTL pages need links to the paper

### ISSUE THREAD

**Issue by maxglycine** [legumeinfo/mine-issues#171]
See PowerPoint slides describing the problems.  I had to click around through a number of different pages concerning a QTL to finally find a page that actually had a link to the paper.  Since this data model of a QTL report has less information about the QTL than the page at SoyBase, we need to allow the user to quickly get to the paper to get the omitted information. @StevenCannon-USDA @jd-campbell @adf-ncgr 
[MineIssues.pptx](https://github.com/user-attachments/files/16730247/MineIssues.pptx)

**Comment by StevenCannon-USDA** [comment]
My take on this:
Making the links to the publication more obvious is important, but there are places where it would be natural to do it and others where it would be really messy to do.

- Slide 1: the text in the Description comes from the README and I suspect wouldn't be straightforward to parse and turn into a link. For one thing, there's no guarantee that the description has a DOI.
- Slide 2: there actually is a link to the paper on that page, under the heading "Publications DOI". I am not sure how the "External Links" section gets populated. Andrew?
- Slide 3: no opinion. Agreed, making the DOI string linkable would be nice if not too difficult; but the "View on doi.org" does work.

**Comment by maxglycine** [comment]
Slide 2  Have you tried clicking that link?  When I did, it did not take me to the paper.  It took me to another mine QTL related
Slide 1  True, not every paper will have a DOI, but it probably has a paper URL.  Maybe the model needs to be changed to have links to both the DOI and the URL.  When there is no DOI, then don't make a link.
 page. @StevenCannon-USDA

**Comment by adf-ncgr** [comment]
We should probably focus on trying to get this into "External Links", as I think the other links are meant to be "Internal Links" (ie clicking any of the links shown in the Publications table, including the one that is a DOI, will just take you to the Glycinemine Publication report page for that publication). I am also not clear on how External Links gets populated but can look into it further. Sounds like the idea is that an "external" link to the paper should be provided not only on the Publications page itself but also on data derived from that Publication (ie the QTL/QTLStudy pages)?

**Comment by maxglycine** [comment]
Yes, I think that making the user go one or two pages into glycine mine just to find a link to the paper is not a good idea.  I saw the words "External Links" and assumed that a link to the paper (an external link) should go there.  If you go to https://mines.legumeinfo.org/glycinemine/report.do?id=391000048 there is a box "QTL Study  Description" and the text and link in that box looks like it would go to the DOI for the paper, but it does not.  It goes to another mine page https://mines.legumeinfo.org/glycinemine/report.do?id=391000011 which has a box "Publications DOI" with a link that looks like a DOI, but it goes to the mine page https://mines.legumeinfo.org/glycinemine/report.do?id=391000002.  This page has a "External Links view this on DOI.org"  that actually goes to the doi for the paper.  If the "External Links" box had a real link to the paper, then all of this clicking around to find the right page with a working link to the paper would not be necessary. It is frustrating to have to do all the clicking around to pages that look a lot alike to find that one "External Link -- DOI.org"

**Comment by StevenCannon-USDA** [comment]
> related Slide 1 True, not every paper will have a DOI, but it probably has a paper URL. Maybe the model needs to be changed to have links to both the DOI and the URL. When there is no DOI, then don't make a link. page. 

@maxglycine - What is in the "Description" field of the QTL pages is up to us. It's plain-text that comes from the README. Sam used the pattern `Further information provided in 10.2135/cropsci2004.0784` for expediency, because there are 341 QTL and GWAS papers. But whatever text goes in the Description, I don't think it's the place to construct an href.

**Comment by maxglycine** [comment]
OK, I am ok with that, if there is an "External Links" that does go to the paper on every QTL report page that does not already have a link to the paper.  Remember, we struggle so they don't have to.  We need to make the UI as user friendly and useful as possible.  If that means changing some of the decisions Sam made then we should do it.

**Comment by adf-ncgr** [comment]
"We struggle so they don't have to" 👍 
FWIW, here's the intermine documentation relevant to configuring "External Links":
http://intermine.org/im-docs/docs/webapp/properties/web-properties#external-links
It's not obvious to me whether this mechanism will support using attributes of related objects (ie the DOI of a publication for the QTLStudy) to shorten the struggle but I will investigate/experiment and let you know.

**Comment by adf-ncgr** [comment]
following up on this a little, if only so the thread doesn't grow stale. I played around a little bit with the configuration of external links, but have so far been unable to find a way to use anything but immediate attributes of the object in the report page to feed to the configuration mechanism. On the other hand, the attributes of associated objects can be added to the top of the report page for the object. For example see:
https://mines.dev.lis.ncgr.org/glycinemine/qtl:0518BW-8_x_0734BW-1.qtl.Oyoo_Benitez_2011:Seed_coat_color_3-1
where the DOI is given in the top section, but the external link is not working properly. 

It may be solvable with some tweaks to the intermine code, but another option would be to use the linkouts mechanism instead. The latter would have the advantage that it would impact the links provided in the trait search web component as well. So, I may try to fiddle with that option instead for now.
